src/test/fuzz/util/descriptor.cpp

```diff
@@ -84,3 +87,34 @@ bool HasDeepDerivPath(const FuzzBufferType& buff, const int max_depth)
     }
     return false;
 }
+
+bool HasLargeFrag(const FuzzBufferType& buff, const int max_subs)
+{
+    std::stack<int> counts;
+    for (const auto& ch: buff) {
+        if (ch == '(') {
+            counts.push(1);
+        } else if (ch == ',' && !counts.empty()) {
+            if (++counts.top() > max_subs) return true;
+        } else if (ch == ')' && !counts.empty()) {
+            counts.pop();
+        }
+    }
+    return false;
+}
+
+bool HasTooManyWrappers(const FuzzBufferType& buff, const int max_wrappers)
+{
+    auto count{0};
+    // We iterate in reverse order to start counting when we encounter a colon.
+    for (const auto& ch: buff | std::views::reverse) {
```

Comment:
Sorry, i saw your comment earlier, just didn't have time to address earlier. Thanks, will push soon.

Reply:
Done (through `.rbegin()` instead as your `std::reverse_iterate` suggestion didn't compile).

Reply:
Changed to use our own `reverse_iterator`. Opted to instanciate the `std::span` from an iterator instead of adding a dependency on https://github.com/bitcoin/bitcoin/pull/30229. Happy to rebase on top of it if it's merged first.

Reply:
Cool, will do since i'll retouch.